aws/data_source_aws_ram_resource_share.go

```diff
@@ -0,0 +1,137 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ram"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsRamResourceShare() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsRamResourceShareRead,
+
+		Schema: map[string]*schema.Schema{
+			"filter": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"values": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+
+			"resource_owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsRamResourceShareRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ramconn
+
+	name := d.Get("name").(string)
+	owner := d.Get("resource_owner").(string)
+
+	filters, filtersOk := d.GetOk("filter")
+
+	params := &ram.GetResourceSharesInput{
+		Name:          aws.String(name),
+		ResourceOwner: aws.String(owner),
+	}
+
+	if filtersOk {
+		params.TagFilters = buildRAMTagFilters(filters.(*schema.Set))
+	}
+
+	for {
+		resp, err := conn.GetResourceShares(params)
+
+		if err != nil {
+			return fmt.Errorf("Error retrieving resource share: empty response for: %s", params)
+		}
+
+		if len(resp.ResourceShares) > 1 {
+			return fmt.Errorf("Multiple resource shares found for: %s", name)
+		}
+
+		if resp == nil || len(resp.ResourceShares) == 0 {
+			return fmt.Errorf("No matching resource found: %s", err)
+		}
+
+		for _, r := range resp.ResourceShares {
+			if aws.StringValue(r.Name) == name {
+				d.SetId(aws.StringValue(r.ResourceShareArn))
+				d.Set("arn", aws.StringValue(r.ResourceShareArn))
+				d.Set("owning_account_id", aws.StringValue(r.OwningAccountId))
+				d.Set("status", aws.StringValue(r.Status))
+				d.Set("tags", r.Tags)
```

Comment:
When using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```go
				if err := d.Set("tags", r.Tags); err != nil {
					return fmt.Errorf("error setting tags: %s", err)
				}
```

After adding this error checking, you should start receiving type errors as `r.Tags` is a `[]*ram.Tag` and not an acceptable type for `d.Set()`. Looking at the existing `aws_ram_resource_share` resource logic, we should be able to borrow its functionality here with `tagsToMapRAM()`:

```suggestion
				if err := d.Set("tags", tagsToMapRAM(r.Tags)); err != nil {
					return fmt.Errorf("error setting tags: %s", err)
				}
```